### PR TITLE
feat: support react-intl >= 3

### PR DESCRIPTION
**Pull request by FRSgit**
remove unnecessary constructors
fix errorprone usage of document.insertBefore - it would always break
add better typings
export ReactIntlPhraseProps so they can be used by end users
upgrade dependencies
fix typescript declaration output path

references #33 & https://trello.com/c/OiWlYaIK

### CODE REVIEW

**Comment by FRSgit** [comment]
Good thing is that the only major change here is a change of `peerDependency`, no other breaking changes. **But** we probably should change this intergration's API to cover regular [`react-intl` api methods](https://github.com/formatjs/react-intl/blob/master/docs/API.md#intlshape)

**Comment by FRSgit** [comment]
Also, should be followed by removal of `state` property from `InjectPhrase` class and upgrade of our [our demo repository](https://github.com/phrase/react-intl-phraseapp-demo) (we could create a typescript demo as well, what do you think @theSoenke)

File: package.json [hunk 1]
```diff
@@ -21,28 +21,28 @@
...
+    "ts-jest": "^25.2.0",
+    "ts-loader": "^6.2.1",
+    "typescript": "^3.7.5",
+    "webpack": "^4.41.6",
+    "webpack-cli": "^3.3.11"
   },
   "peerDependencies": {
-    "react": ">= 15.5.4",
-    "react-intl": ">= 2.2.3 < 3"
+    "react": ">=16.12.0",
```
Comment (theSoenke) [inline package.json]: Isn't this too restrict? Is there any issue with older releases?

**Comment by theSoenke** [comment]
@FRSgit Upgrading the demo app sounds good. But I think I would not create an extra TypeScript demo for now. Does not provide a lot of additional benefits to also have a TypeScript version I would say

File: package.json (hunk 1 above)
Comment (FRSgit) [reply package.json]: You're right. Just checked and `react-intl` has peerDependency of `react` with version `^16.3.0`, so we should go with the same.
Another question would be - do we need to specify `react` peerDependency at all, when we have `react-intl` which should always require `react` itself?

**Comment by FRSgit** [comment]
@theSoenke Yes and no, I just found out that our typings weren't working for last release, so it might be beneficial. Also, I've just created exported props type which can easily be used to correctly type react component by adding our props to the type definition, but I don't have a demo to show how it could be implemented :<

File: package.json (hunk 1 above)
Comment (theSoenke) [reply package.json]: I don't really have a strong opinion on this. Remove it if this feels right to you :)

Comment (FRSgit) [reply package.json]: We have the `react` required in our codebase, so maybe it's better to leave it here, I'll just upgrade the version number

Comment (FRSgit) [reply package.json]: Done!

**Pull request merged**